### PR TITLE
Add sha2-style base64 crypt format

### PR DESCRIPTION
**Pull request by ibotty**
This replaces #738.  It is minimally invasive.

### CODE REVIEW

File: base64ct/src/alphabet.rs [hunk 1]
```diff
@@ -70,7 +71,7 @@ pub trait Alphabet: 'static + Copy + Debug + Eq + Send + Sized + Sync {
         ret
     }
 
-    /// Encode 3-bytes of a Base64 message.
+    // Encode 3-bytes of a Base64 message.
```
Comment (tarcieri) [inline base64ct/src/alphabet.rs:74]: This seems unnecessary

**Review by tarcieri — APPROVED** [review]
One nit if you care to fix it, otherwise I can merge as-is

File: base64ct/src/alphabet.rs (hunk 1 above)
Comment (ibotty) [reply base64ct/src/alphabet.rs]: Oh, I missed that.  Sorry.  Fixed in latest commit.

**Comment by ibotty** [comment]
Thank you!

**Pull request merged**